File: app/src/main/java/com/example/despensacx/MesAdapter.java
package com.example.despensacx;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import java.util.Map;

public class MesAdapter extends RecyclerView.Adapter<MesAdapter.MesViewHolder> {

    private final List<EstadisticaModel.MesModel> listaMeses;

    public MesAdapter(List<EstadisticaModel.MesModel> listaMeses) {
        this.listaMeses = listaMeses;
    }

    @NonNull
    @Override
    public MesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_estadistica_mes, parent, false);
        return new MesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MesViewHolder holder, int position) {
        EstadisticaModel.MesModel item = listaMeses.get(position);
        holder.tvMesNombre.setText(item.getMesNombre());
        holder.tvTotalMes.setText(FormatUtils.formatCurrency(item.getTotalMes()));

        // Desglose por Tiendas
        holder.containerTiendas.removeAllViews();
        Map<String, Double> gastosTienda = item.getGastosPorTienda();

        if (gastosTienda != null && !gastosTienda.isEmpty()) {
            for (Map.Entry<String, Double> entry : gastosTienda.entrySet()) {
                View row = LayoutInflater.from(holder.itemView.getContext())
                        .inflate(R.layout.item_desglose_tienda, holder.containerTiendas, false);

                TextView tvNombreTienda = row.findViewById(R.id.tvNombreTienda);
                TextView tvMontoTienda = row.findViewById(R.id.tvMontoTienda);

                tvNombreTienda.setText(entry.getKey());
                tvMontoTienda.setText(FormatUtils.formatCurrency(entry.getValue()));

                holder.containerTiendas.addView(row);
            }
        }
    }

    @Override
    public int getItemCount() {
        return listaMeses != null ? listaMeses.size() : 0;
    }

    static class MesViewHolder extends RecyclerView.ViewHolder {
        TextView tvMesNombre, tvTotalMes;
        LinearLayout containerTiendas;

        public MesViewHolder(@NonNull View itemView) {
            super(itemView);
            tvMesNombre = itemView.findViewById(R.id.tvMesNombre);
            tvTotalMes = itemView.findViewById(R.id.tvTotalMes);
            containerTiendas = itemView.findViewById(R.id.containerTiendas);
        }
    }
}

File: app/src/main/kotlin/com/example/despensacx/adapter/MesAdapter.kt
package com.example.despensacx.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.despensacx.R
import com.example.despensacx.model.EstadisticaModel
import com.example.despensacx.utils.FormatUtils

class MesAdapter(private val listaMeses: List<EstadisticaModel.MesModel>) :
    RecyclerView.Adapter<MesAdapter.MesViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MesViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_estadistica_mes, parent, false)
        return MesViewHolder(view)
    }

    override fun onBindViewHolder(holder: MesViewHolder, position: Int) {
        val item = listaMeses[position]
        holder.tvMesNombre.text = item.mesNombre
        holder.tvTotalMes.text = FormatUtils.formatCurrency(item.totalMes)

        // Desglose por Tiendas
        holder.containerTiendas.removeAllViews()
        val gastosTienda = item.gastosPorTienda

        if (gastosTienda.isNotEmpty()) {
            for ((key, value) in gastosTienda) {
                val row = LayoutInflater.from(holder.itemView.context)
                    .inflate(R.layout.item_desglose_tienda, holder.containerTiendas, false)

                val tvNombreTienda = row.findViewById<TextView>(R.id.tvNombreTienda)
                val tvMontoTienda = row.findViewById<TextView>(R.id.tvMontoTienda)

                tvNombreTienda.text = key
                tvMontoTienda.text = FormatUtils.formatCurrency(value)

                holder.containerTiendas.addView(row)
            }
        }
    }

    override fun getItemCount(): Int = listaMeses.size

    class MesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvMesNombre: TextView = itemView.findViewById(R.id.tvMesNombre)
        val tvTotalMes: TextView = itemView.findViewById(R.id.tvTotalMes)
        val containerTiendas: LinearLayout = itemView.findViewById(R.id.containerTiendas)
    }
}
